### PR TITLE
Dropped Tag from AssetClass types

### PR DESCRIPTION
**Pull request by SeungheonOh**

### CODE REVIEW

**Review by kozross — APPROVED** [review]

**Comment by SeungheonOh** [comment]
Please don't merge this yet. I'm working on liqwid side simultaneously and I might need to add some more utilities.

**Comment by SeungheonOh** [comment]
PR is ready to go. 

Liqwid-onchain has PR ready for this PR. https://github.com/mlabs-haskell/liqwid-onchain/pull/179

File: src/Plutarch/Extra/AssetClass.hs [hunk 1]
```diff
@@ -69,7 +69,7 @@ import qualified PlutusTx
 
  @since 3.9.0
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/AssetClass.hs:70]: needs a version bump, since the type changed, right?

File: src/Plutarch/Extra/AssetClass.hs [hunk 2]
```diff
@@ -94,13 +94,30 @@ data AssetClass (unit :: Symbol) = AssetClass
       FromJSONKey
     , -- | @since 3.9.0
       ToJSONKey
+    , -- | @since 3.10.0
+      SOP.Generic
+    )
+  deriving
+    ( -- | @since 3.9.0
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/AssetClass.hs:101]: versions for these derivations should be 3.10.0

File: src/Plutarch/Extra/AssetClass.hs [hunk 3]
```diff
@@ -94,13 +94,30 @@ data AssetClass (unit :: Symbol) = AssetClass
...
+            AssetClass
+            PAssetClassData
+        )
 
 {- | A Scott-encoded Plutarch equivalent to 'AssetClass'.
 
  @since 3.9.0
 -}
-data PAssetClass (unit :: Symbol) (s :: S) = PAssetClass
+data PAssetClass (s :: S) = PAssetClass
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/AssetClass.hs:120]: version bump

File: src/Plutarch/Extra/AssetClass.hs [hunk 4]
```diff
@@ -118,35 +135,61 @@ data PAssetClass (unit :: Symbol) (s :: S) = PAssetClass
     )
 
 -- | @since 3.9.0
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/AssetClass.hs:137]: update since

File: src/Plutarch/Extra/AssetClass.hs [hunk 5]
```diff
@@ -118,35 +135,61 @@ data PAssetClass (unit :: Symbol) (s :: S) = PAssetClass
...
+
+-- | @since 3.10.0
+passetClassT ::
+  forall (unit :: Symbol) (s :: S).
+  Term s (PCurrencySymbol :--> PTokenName :--> PTagged unit PAssetClass)
+passetClassT = phoistAcyclic $
+  plam $ \sym tk ->
+    ppure #$ passetClass # sym # tk
+
 -- | @since 3.9.0
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/AssetClass.hs:157]: update since

File: src/Plutarch/Extra/Value.hs [hunk 6]
```diff
@@ -187,14 +227,27 @@ padaOf = phoistAcyclic $
...
-  AssetClass unit ->
+  forall (keys :: KeyGuarantees) (amounts :: AmountGuarantees) (s :: S).
+  AssetClass ->
   Term s (PValue keys amounts :--> PInteger)
 passetClassValueOf' (AssetClass sym token) =
   phoistAcyclic $
     plam $ \value ->
       Value.pvalueOf # value # pconstant sym # pconstant token
 
+{- | Tagged version of `passetClassValueOf'`.
```
Comment (peter-mlabs) [inline src/Plutarch/Extra/Value.hs:238]: use quotes here to get haddock linking

**Review by peter-mlabs — CHANGES_REQUESTED** [review]
looks good overall.

Need to updated some `since`s, especially where you only deleted the tag.

Also, in some haddock comments where you've written `` `foo` ``, it should be `"foo"`, so that we can get haddock linking.

**Comment by kozross** [comment]
@peter-mlabs It's actually 'foo'. Single quotes create Haddock links.

**Review by peter-mlabs — APPROVED** [review]
Looks good now, thanks @SeungheonOh!

**Pull request merged**